walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,121 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    public static final int START_OF_DAY = TimeRange.START_OF_DAY;
+    public static final int END_OF_DAY = TimeRange.END_OF_DAY;
+    public static final TimeRange WHOLE_DAY = TimeRange.WHOLE_DAY;
+    public static final Comparator<TimeRange> ORDER_BY_START = TimeRange.ORDER_BY_START;
+
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+
+        // Array of queries represenitng times to meet.
+        List<TimeRange> query = new ArrayList<>();
+        
+        // Integer represenitng the number of events.
+        int numberOfEvents = events.size();
+        
+        // Integer representing the Start of a work day.
+        int timeOfStart = START_OF_DAY;
+
+        // Integer representing the End of a work day.
+        int timeOfEnd = END_OF_DAY;
+        
+        // Integer represenig the duration of a meeting request.
+        int desiredDuration = (int) request.getDuration();
+        
+        //  Event counter used whilst iterating.
+        int eventCounter = 0;
+        
+        int event_start = 0;
+        int event_end = 0;
+
+        // Set of stirngs representing unique employees set to attend this meeting.
+        Set<String> event_attendees = null;
+
+        // Collection of string reprenting who is to attend the requedted meeting
+        Collection<String> request_attendees = request.getAttendees();
+        
+        // Boolean used to indictae whter or not the attendee should be ignored.
+        boolean shouldIgnore = true;
+
+        // Check if duration of event is longer than a whole day and if so return empty.
+        if ((int)request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
```

Comment:
is the int cast necessary here?

Reply:
I believe so, since the getDuration function returns a long